test/test-functions

```diff
@@ -512,6 +512,9 @@ run_qemu() {
         read -ra user_qemu_options <<< "$QEMU_OPTIONS"
         qemu_options+=("${user_qemu_options[@]}")
     fi
+    if [[ -n "${QEMU_OPTIONS_ARRAY+x}" ]]; then
+        qemu_options+=("${QEMU_OPTIONS_ARRAY[@]}")
+    fi
```

Comment:
@mrc0mmand Do you think that this is a good way to add options with space?

Reply:
I think there is no need to check if the array is empty. `"${QEMU_OPTIONS_ARRAY[@]}"` expands to nothing if it's empty, and the `+=` becomes a no-op.

Reply:
(And this is the correct way I think, as this is expanded per-element.)

Reply:
Problem here is that `QEMU_OPTIONS_ARRAY` may not be declared, and if not, `"${QEMU_OPTIONS_ARRAY[@]}"` triggers error, as we usually call `set -u`. Hence, we need to check if it is declared.

Not familiar, but `[[ -v QEMU_OPTIONS_ARRAY ]]` may be better to check that. WDYT?

Reply:
Oh, right.

Then `-v` would be the way to go :)

Reply:
It's unfortunate we have to do this dance, but I guess it will do. I thought about teaching `run_qemu()` to check if `$QEMU_OPTIONS` is an array, and if so skip the intermediate conversion, but that introduces crapload of other issues.

But to make this simpler, you can use just

```suggestion
    qemu_options+=(${QEMU_OPTIONS_ARRAY:+"${QEMU_OPTIONS_ARRAY[@]}"})
```

Reply:
(With newer bash versions you can do just `qemu_options+=("${QEMU_OPTIONS_ARRAY[@]}")` and it will do the right thing even with `set -u`, but versions like 4.4 and lower treat expansion even of a defined, but empty array as an unbound variable, so it's better to stay on the safe side)

Reply:
@mrc0mmand Oh, that's much simpler. Thanks.